internal/net/conn_other.go

```diff
@@ -0,0 +1,58 @@
+//go:build test
+
+package net
+
+import (
+	"context"
+
+	"github.com/drand/drand/v2/internal/metrics"
+	"go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/connectivity"
+	insecure "google.golang.org/grpc/credentials/insecure"
+)
+
+// conn retrieve an already existing conn to the given peer or create a new one
+func (g *grpcClient) conn(ctx context.Context, p Peer) (*grpc.ClientConn, error) {
+	g.Lock()
```

Comment:
a big warning log that "NON-TLS CONNECTION FOR TEST ONLY" just in case we somehow build the wrong one would be good heh